### DISCUSSION TITLE
Export by selectedRowsQuery

### DISCUSSION THREAD

**Discussion by 22syamsudar** [rappasoft/laravel-livewire-tables#260] [Q&A]
if ($this->selectedRowsQuery->count() > 0) {
          return (new UserExport($this->selectedRowsQuery))->download($this->tableName.'.xlsx');
      }

in my UserExport like this below  

namespace App\Exports;

use App\Models\UserExport;
use Illuminate\Contracts\Support\Responsable;
use Maatwebsite\Excel\Concerns\FromQuery;
use Maatwebsite\Excel\Concerns\Exportable;  

class UserExport implements FromQuery, Responsable
{
    use Exportable; 
    protected $id;  
    public function __construct($id)
    {
        $this->id = $id;
    }

    public function query()
    {    
        return UserExport::query()->whereIn('id',$this->id);
    } 
}

then it return
SQLSTATE[21000]: Cardinality violation: 1241 Operand should contain 1 column(s) (SQL: select * from `users` where `id` in (select * from `users` where `status` = 1 and `id` in (15) order by `nama` asc) order by `users`.`id` asc limit 1000 offset 0)

and it's not exported...
any suggestion please?

**Comment by rappasoft** [comment]
This is not an issue with the package, but I'll give you an example from one of my projects anyway:

```php
public function exportSelected()
    {
        if ($this->selectedRowsQuery->count() > 0) {
            return (new UserExport($this->selectedRowsQuery))->download($this->tableName.'.xlsx');
        }

        $this->notify(__('You did not select any users to export.'), 'danger');
    }
```

UserExport.php

```php
<?php

namespace App\Domains\User\Exports;

use Illuminate\Database\Eloquent\Builder;
use Maatwebsite\Excel\Concerns\Exportable;
use Maatwebsite\Excel\Concerns\FromQuery;
use Maatwebsite\Excel\Concerns\WithHeadings;
use Maatwebsite\Excel\Concerns\WithMapping;

/**
 * Class UserExport
 *
 * @package App\Domains\User\Exports
 */
class UserExport implements FromQuery, WithMapping, WithHeadings
{
    use Exportable;

    /**
     * @var Builder
     */
    public Builder $builder;

    /**
     * UserExport constructor.
     *
     * @param  Builder  $builder
     */
    public function __construct(Builder $builder)
    {
        $this->builder = $builder;
    }

    /**
     * @return Builder|\Illuminate\Database\Query\Builder
     */
    public function query()
    {
        return $this->builder;
    }

    /**
     * @return string[]
     */
    public function headings(): array
    {
        return [
            'Name',
            'E-mail',
            'User Type',
            'Active',
        ];
    }

    /**
     * @param  mixed  $row
     *
     * @return array
     */
    public function map($row): array
    {
        return [
            $row->name,
            $row->email,
            ucfirst($row->type),
            $row->active === true ? 'Yes' : 'No',
        ];
    }
}
```

**Reply by 22syamsudar** [reply]
thanks man... you are amazing...
the others will find this solution...
we appreciate it...
